static/js/components/Periodogram.jsx

```diff
@@ -141,11 +154,26 @@ function GLS(t_data_uf, y_data_uf, kwa) {
   const wy = dotMultiply(w, y);
   const YY = dot(wy, y); // Eq.(10), variance for the zero mean data
 
-  const df = 1 / tbase / ofac; // frequency sampling depends on the time span, default for start frequency
-  const delta_t = tbase / (nt - 1); // mean time sampling
-  const fbeg = parseFloat(kwargs.fbeg || df);
-  const fend = parseFloat(kwargs.fend || (0.5 * ofac) / delta_t);
-  const nf = Math.floor((fend - fbeg) / df) + 1; // size of frequency grid
+  // frequency sampling depends on the time span, default for start frequency
+  let df = 1 / tbase / ofac;
+  // chose as delta_t which is the min of mean time sampling, 75%-tile of the time differences, or fixed low-end
+  const delta_t = Math.min(
+    tbase / (nt - 1),
+    quantileSeq(diffs, 0.75, false),
+    0.05 * ofac
+  );
+  let fbeg = parseFloat(kwargs.fbeg || (df * ofac) / 2);
+  let fend = parseFloat(kwargs.fend || (0.5 * ofac) / delta_t);
+  let nf = Math.floor((fend - fbeg) / df) + 1; // size of frequency grid
+
+  // ensure that we're only doing as many as 300,000 calculations (~<5 seconds)
+  // bump down the frequency binning (df) and shrink the start and ending limits
+  while (nf > 3e5) {
```

Comment:
Until we have a client-side graphics card implementation, limit the total number of frequencies to 300,000. This takes at most ~5 seconds on a modern system/browser.